hopp/simulation/technologies/tidal/mhk_tidal_plant.py

```diff
@@ -0,0 +1,235 @@
+
+        self.mhk_costs.simulate_costs()
+        cost_dict = self.mhk_costs.cost_outputs
+
+        capex = cost_dict['structural_assembly_cost_modeled']+\
+            cost_dict['power_takeoff_system_cost_modeled']+\
+            cost_dict['mooring_found_substruc_cost_modeled']
+        bos = cost_dict['development_cost_modeled']+\
+            cost_dict['eng_and_mgmt_cost_modeled']+\
+            cost_dict['plant_commissioning_cost_modeled']+\
+            cost_dict['site_access_port_staging_cost_modeled']+\
+            cost_dict['assembly_and_install_cost_modeled']+\
+            cost_dict['other_infrastructure_cost_modeled']
+        elec_infrastruc_costs = cost_dict['array_cable_system_cost_modeled']+\
+            cost_dict['export_cable_system_cost_modeled']+\
+            cost_dict['onshore_substation_cost_modeled']+\
+            cost_dict['offshore_substation_cost_modeled']+\
+            cost_dict['other_elec_infra_cost_modeled']
+        financial = cost_dict['project_contingency']+\
+            cost_dict['insurance_during_construction']+\
+            cost_dict['reserve_accounts']
+
+        total_installed_cost = capex+bos+elec_infrastruc_costs+financial
+        
+        return self._financial_model.value("total_installed_cost", total_installed_cost)
+
+    def system_capacity_by_num_devices(self, tidal_size_kw: float):
+        """
+        Sets the system capacity by adjusting the number of devices
+        """
+        new_num_devices = round(tidal_size_kw / self.device_rated_power)
+        if self.number_devices != new_num_devices:
+            self.number_devices = new_num_devices
+
+    def simulate(self, interconnect_kw: float, project_life: int = 25, lifetime_sim=False):
+        """
+        Run the system and financial model
+
+        Args:
+            interconnect_kw: grid interconnect
+            project_life: Number of years in the analysis period (expected
+                project lifetime)
+            lifetime_sim:
+                For simulation modules which support simulating each year of the
+                    project_life, whether or not to do so; otherwise the first year
+                    data is repeated
+        """
+
+        self.calculate_total_installed_cost()
+        super().simulate(interconnect_kw, project_life)
+
+    @property
+    def device_rated_power(self) -> float:
+        return self._system_model.device_rated_power
+
+    @device_rated_power.setter
+    def device_rated_power(self, device_rate_power: float):
+        self._system_model.device_rated_power = device_rate_power
+        if self.mhk_costs != None:
+            self.mhk_costs.device_rated_power = device_rate_power
+
+    @property
+    def number_devices(self) -> int:
+        return self._system_model.MHKTidal.number_devices
+
+    @number_devices.setter
+    def number_devices(self, number_devices: int):
+        self._system_model.MHKTidal.number_devices = number_devices
+        if self.mhk_costs != None:
```

Comment:
Non-blocking nitpick that requires no change here: I'd suggest using `is not None` instead of `!= None` as None is a singleton. Not that important and using `!=` doesn't usually have negative impacts, but I was caught by an edge case once so I like to suggest it. https://stackoverflow.com/a/2209781/5050416 for more info 
(I know this occurs throughout HOPP but seeing it here made me want to mention it)